src/app/modules/main/wallet_section/collectibles/module.nim

```diff
@@ -81,49 +81,36 @@ method currentCollectibleModuleDidLoad*(self: Module) =
   self.checkIfModuleDidLoad()
 
 method fetchOwnedCollectibles*(self: Module) =
-  self.controller.fetchOwnedCollectibles(self.chainId, self.address)
+  self.controller.fetchOwnedCollectibles(self.chainId, self.addresses)
 
 method filterChanged*(self: Module, addresses: seq[string], chainIds: seq[int]) =
   let network = self.controller.getNetwork()
-  let account = self.controller.getWalletAccountByAddress(addresses[0])
   self.chainId = network.chainId
-  self.address = account.address
-  self.currentCollectibleModule.setCurrentAddress(network, self.address)
+  self.addresses = addresses
+  self.currentCollectibleModule.setCurrentNetwork(network)
+  self.view.setCollectibles(@[])
+  let data = self.controller.getOwnedCollectibles(self.chainId, self.addresses)
 
-  let data = self.controller.getOwnedCollectibles(self.chainId, self.address)
+  if not data[0].anyLoaded:
+    self.fetchOwnedCollectibles()
 
-  # Trigger a fetch the first time we switch to an account
-  if not data.anyLoaded:
-    self.controller.fetchOwnedCollectibles(self.chainId, self.address)
-
-  self.setCollectibles(self.chainId, self.address, data)
+  self.setCollectibles(data)
 
 proc ownedCollectibleToItem(self: Module, oc: OwnedCollectible): Item =
   let c = self.controller.getCollectible(self.chainId, oc.id)
   let col = self.controller.getCollection(self.chainId, c.collectionSlug)
   return collectibleToItem(c, col, oc.isFromWatchedContract)
 
 method onFetchStarted*(self: Module, chainId: int, address: string) =
-  if self.chainId == chainId and self.address == address:
+  if self.chainId == chainId and address in self.addresses:
     self.view.setIsFetching(true)
 
-method setCollectibles*(self: Module, chainId: int, address: string, data: CollectiblesData) =
```

Comment:
`setCollectible`s did not care about how many collectibles were last fetched by the collectibles service, that's why I used this for the initialization (just set the View's state to whatever the service has), and then `appendCollectibles` for the updates (just add the new collectibles to the list). I'd love to be able to only use `setCollectibles` whenever the service has a new state, but resetting the model makes the scroll position to go to the top.